nisystemlink/clients/systemsstate/_systems_state_client.py

```diff
@@ -0,0 +1,256 @@
+
+    @get(
+        "states",
+        args=[
+            Query("Skip"),
+            Query("Take"),
+            Query("Workspace"),
+            Query("Architecture"),
+            Query("Distribution"),
+        ],
+    )
+    def get_states(
+        self,
+        skip: Optional[int] = None,
+        take: Optional[int] = None,
+        workspace: Optional[str] = None,
+        architecture: Optional[models.Architecture] = None,
+        distribution: Optional[models.Distribution] = None,
+    ) -> models.StateDescriptionListResponse:
+        """Get the list of states
+
+        Args:
+            `skip`: Number of files to skip in the result when paging
+            `take`: Number of files to return in the result
+            `workspace`: The workspace id of the states
+            `architecture`: The states that are compatible with the requested architecture
+            `distribution`: The states that are compatible with the requested distribution
+
+        Returns:
+            The list of states for the given query parameters
+
+        Raises:
+            ApiException : if unable to communicate with the ``/nisystemsstate`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @post("states", args=[Body])
+    def create_state(self, new_state: models.StateRequest) -> models.StateResponse:
+        """Create state
+
+        Args:
+            `state`: Information about the state to be created.
+
+        Returns:
+            The state created
+
+        Raises:
+            ApiException : if unable to communicate with the ``/nisystemsstate`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @post("export-state", args=[Body])
+    def export_state(self, export_state_request: models.ExportStateRequest) -> str:
+        """Generate state export
+
+        Args:
+            `export_state_request`: Contains identifying information on the state to export.
+
+        Returns:
+            The exported state as a file.
+
+        Raises:
+            ApiException : if unable to communicate with the ``/nisystemsstate`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @post("export-state-from-system", args=[Body])
+    def export_state_from_system(
+        self, export_state_from_system_request: models.ExportStateFromSystemRequest
+    ) -> str:
+        """Generate state export of a system
+
+        Args:
+            `export_state_from_system_request`: Contains the system id from which to export the state.
+
+        Returns:
+            The exported state of the system as a file.
+
+        Raises:
+            ApiException : if unable to communicate with the ``/nisystemsstate`` Service
+                or provided an invalid argument.
+        """
+        ...
+
+    @get("states/{stateId}", args=[Path("stateId")])
+    def get_state(self, state_id: str) -> models.StateResponse:
```

Comment:
The method name and arg `id` together implies `state_id`
```suggestion
    def get_state(self, id: str) -> models.StateResponse:
```